modeshape-jcr/src/main/java/org/modeshape/jcr/value/binary/Database.java

```diff
@@ -155,11 +159,29 @@ protected Database( Connection connection,
         }
         this.prefix = prefix == null ? null : prefix.trim();
         this.tableName = this.prefix != null && this.prefix.length() != 0 ? this.prefix + TABLE_NAME : TABLE_NAME;
+        this.maxExtractedTextLength = determineMaxExtractedTextLength(metaData);
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Using max length for extracted text '{0}'", maxExtractedTextLength);
+        }
 
         initializeStatements();
         initializeStorage(connection);
     }
 
+    private int determineMaxExtractedTextLength( DatabaseMetaData metaData ) {
+        ResultSet resultSet = null;
+        try {
+            resultSet = metaData.getColumns(null, null, null,  EXTRACTED_TEXT_COLUMN_NAME);
+            return resultSet.next() ? resultSet.getInt("COLUMN_SIZE") : DEFAULT_MAX_EXTRACTED_TEXT_LENGTH;
```

Comment:
the only additional filtering criteria that's available is "tableName". However, using & passing that as a simple string (param) yields no result (i.e. no columns found) on certain DBs. For example: H2.

I'm happy to take any pointers as to how that param should be used with `dbMetadata`.